src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp

```diff
@@ -0,0 +1,42 @@
+#include <Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.h>
+#include <Functions/JSONPath/ASTs/ASTJSONPathMemberAccess.h>
+
+#include <Parsers/Lexer.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/ASTIdentifier.h>
+
+namespace DB
+{
+
+/**
+ *
+ * @param pos token iterator
+ * @param node node of ASTJSONPathMemberAccess
+ * @param expected stuff for logging
+ * @return was parse successful
+ */
+bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    auto member_access = std::make_shared<ASTJSONPathMemberAccess>();
+    node = member_access;
```

Comment:
Because node is of type ASTPtr, so we would not be able to treat it as an object of type ASTJSONPathMemberAccess (for instance, we would not be able access its members)

Reply:
You need to use `typeid_cast` for such cases.

Reply:
Actually, judging by ClickHouse core query parsers, the way we implemented it is preferable

https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/ParserTablePropertiesQuery.cpp#L14
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/ParserAlterQuery.cpp#L22
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/ParserAlterQuery.cpp#L755
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/ParserCase.cpp#L88
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/ParserCreateQuery.h#L65
and so on

Reply:
ok